Storage/Bookmarks/Bookmarks.swift

```diff
@@ -23,6 +23,7 @@ public protocol SyncableBookmarks: class, ResettableSyncStorage, AccountRemovalD
     func treeForMirror() -> Deferred<Maybe<BookmarkTree>>
     func applyLocalOverrideCompletionOp(_ op: LocalOverrideCompletionOp, itemSources: ItemSources) -> Success
     func applyBufferUpdatedCompletionOp(_ op: BufferUpdatedCompletionOp) -> Success
+    func getServerRecordsCount() -> Deferred<Maybe<Int>>
```

Comment:
This isn't really a count of server records — it's a count of aggregate remote state.

Consider, for example, that someone can `HTTP DELETE` records from the server, and Sync clients will never see those omissions until they drop all local state and redownload.

I suggest here something like `getUpstreamRecordCount` or `getNonLocalRecordCount`.